## checking 3' end read_SRR23849628.535194 for adapter

Is that adapter sequences that is seen after what looks to correspond to the poly-A tail in read @SRR23849628.535194 from `USP7_locus_mapped_SRX19662564_reads.fastq`?   
Just a quick assessment to see if it, or somewhat different versions given nanopore reads can be error prone, occur often in the reads.

This notebook run in sessions launched from blast-binder because it has PatMatch installed in sessions, could also have used patmatch-binder sessions. ( [my blast-binder repo](https://github.com/fomightez/blast-binder) or [my patmatch-binder repo](https://github.com/fomightez/patmatch-binder))

In [2]:
%%bash
awk '{if(NR%4==1) {printf(">%s\n",substr($0,2));} else if(NR%4==2) print;}' USP7_locus_mapped_SRX19662564_reads.fastq > output.fasta

In [7]:
!curl -O https://raw.githubusercontent.com/fomightez/sequencework/master/patmatch-utilities/patmatch_results_to_df.py

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 18734  100 18734    0     0  91427      0 --:--:-- --:--:-- --:--:-- 91833


In [12]:
%%capture
adapter_candidate = 'CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA'
#!fqgrep --reverse-complement --color auto {USP7_TSL4_RET_I} USP7_locus_mapped_SRX19662564_reads.fastq

pm_result = !perl patmatch_1.2/patmatch.pl -c {adapter_candidate} output.fasta 6 ids
from patmatch_results_to_df import patmatch_results_to_df
my_pattern= "CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA"
df = patmatch_results_to_df(pm_result.n, pattern=my_pattern, name="3prime_end_of_read")

In [13]:
df

FASTA_id  hit_number                  hit_id start   end  \
0    SRR23849628.1017513           1    3prime_end_of_read-1  2252  2292   
1    SRR23849628.1032863           1    3prime_end_of_read-2  1659  1699   
2    SRR23849628.1037350           1    3prime_end_of_read-3  1970  2009   
3    SRR23849628.1066851           1    3prime_end_of_read-4  5808  5844   
4    SRR23849628.1098930           1    3prime_end_of_read-5  2062  2097   
..                   ...         ...                     ...   ...   ...   
357   SRR23849628.870297           1  3prime_end_of_read-358  2950  2989   
358   SRR23849628.902956           1  3prime_end_of_read-359  1976  2016   
359    SRR23849628.90551           1  3prime_end_of_read-360   451   490   
360   SRR23849628.919056           1  3prime_end_of_read-361  2007  2046   
361   SRR23849628.971705           1  3prime_end_of_read-362  1214  1250   

     strand                           matching pattern  \
0         1  GTAGATTGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
1         1  CGGAGATCGGAAGAGCGTCGTGTAGAGCAATACGTAATGGA   
2         1   TAGATTGGAAGAGTGTTGTGTAGAGCAATACGTAACTGAA   
3         1      CGGAGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
4         1       TTGCAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
..      ...                                        ...   
357       1   CAAGATCGGAGAGTGTCGTGTAGAGCAATACGTAACTGAA   
358       1  GATAATTGCAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
359       1   GTAGATCTAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
360       1   GTATTGGAAGAGTGTCTGTGTAGAGCAATACGTAACTGAA   
361       1      GTTATCGGAAGCGTCGTGTAGAGCAATACGTAACGAA   

                                  query pattern  
0    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
1    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
2    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
3    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
4    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
..                                          ...  
357  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
358  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
359  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
360  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
361  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  

[362 rows x 8 columns]

Make sure no duplicates. Just keep first, if there is.

In [15]:
dedup_df = df.drop_duplicates(subset=['FASTA_id'], keep='first')
dedup_df

FASTA_id  hit_number                  hit_id start   end  \
0    SRR23849628.1017513           1    3prime_end_of_read-1  2252  2292   
1    SRR23849628.1032863           1    3prime_end_of_read-2  1659  1699   
2    SRR23849628.1037350           1    3prime_end_of_read-3  1970  2009   
3    SRR23849628.1066851           1    3prime_end_of_read-4  5808  5844   
4    SRR23849628.1098930           1    3prime_end_of_read-5  2062  2097   
..                   ...         ...                     ...   ...   ...   
357   SRR23849628.870297           1  3prime_end_of_read-358  2950  2989   
358   SRR23849628.902956           1  3prime_end_of_read-359  1976  2016   
359    SRR23849628.90551           1  3prime_end_of_read-360   451   490   
360   SRR23849628.919056           1  3prime_end_of_read-361  2007  2046   
361   SRR23849628.971705           1  3prime_end_of_read-362  1214  1250   

     strand                           matching pattern  \
0         1  GTAGATTGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
1         1  CGGAGATCGGAAGAGCGTCGTGTAGAGCAATACGTAATGGA   
2         1   TAGATTGGAAGAGTGTTGTGTAGAGCAATACGTAACTGAA   
3         1      CGGAGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
4         1       TTGCAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
..      ...                                        ...   
357       1   CAAGATCGGAGAGTGTCGTGTAGAGCAATACGTAACTGAA   
358       1  GATAATTGCAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
359       1   GTAGATCTAAGAGCGTCGTGTAGAGCAATACGTAACTGAA   
360       1   GTATTGGAAGAGTGTCTGTGTAGAGCAATACGTAACTGAA   
361       1      GTTATCGGAAGCGTCGTGTAGAGCAATACGTAACGAA   

                                  query pattern  
0    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
1    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
2    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
3    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
4    CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
..                                          ...  
357  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
358  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
359  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
360  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  
361  CGTTGATCGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA  

[362 rows x 8 columns]

Same result, and so there were no duplicates.

So 362 out of the 863 have match to that sequence and so I'm guessing it is an adapter of some sort.  
I could explore this more by looking further at position of each pattern match in the sequence, but the first one I looked at , other than the original from `@SRR23849628.535194`, using 'find' in a text editor `GTAGATTGGAAGAGCGTCGTGTAGAGCAATACGTAACTGAA` from `SRR23849628.1017513` showed it was in four different ones at the end. This, I suspect the trend holds. Seems sufficient to support it being an adapter sequence of some sort.

This is just meant to argue this is the case as I chose 6 for the `ids` value to give a lot of flexibility but not be too far off. It wasn't meant to be thorough, but just argue it is safe to consider that technical and not biological.